akka-actor/src/main/scala/akka/util/LineNumbers.scala

```diff
@@ -0,0 +1,320 @@
+        case ((low, high), (start, end)) ⇒ (Math.min(low, start), Math.max(high, end))
+      } match {
+        case (Int.MaxValue, 0) ⇒ None
+        case other             ⇒ Some(other)
+      }
+    } else {
+      dbg(s"  (skipped)")
+      for (_ ← 1 to count) skipMethodOrField(d)
+      None
+    }
+  }
+
+  private def readMethod(d: DataInputStream, code: Int, lnt: Int, filter: Option[String])(implicit c: Constants): Option[(Int, Int)] = {
+    skip(d, 2) // access flags
+    val name = d.readUnsignedShort() // name
+    skip(d, 2) // signature
+    dbg(s"  ${c(name)}")
+    val attributes =
+      for (_ ← 1 to d.readUnsignedShort()) yield {
+        val tag = d.readUnsignedShort()
+        val length = d.readInt()
+        if (tag != code || (filter.isDefined && c(name) != filter.get)) {
+          skip(d, length)
+          None
+        } else {
+          skip(d, 4) // shorts: max stack, max locals
+          skip(d, d.readInt()) // skip byte-code
+          // skip exception table: N records of 4 shorts (start PC, end PC, handler PC, catch type)
+          skip(d, 8 * d.readUnsignedShort())
+          val possibleLines =
+            for (_ ← 1 to d.readUnsignedShort()) yield {
+              val tag = d.readUnsignedShort()
+              val length = d.readInt()
+              if (tag != lnt) {
+                skip(d, length)
+                None
+              } else {
+                val lines =
+                  for (_ ← 1 to d.readUnsignedShort()) yield {
+                    skip(d, 2) // start PC
+                    d.readUnsignedShort() // finally: the line number
+                  }
+                Some(lines.min -> lines.max)
+              }
+            }
+          dbg(s"    nested attributes yielded: $possibleLines")
+          possibleLines.flatten.headOption
+        }
+      }
+    attributes.flatten.headOption
+  }
+
+  private def readAttributes(d: DataInputStream)(implicit c: Constants): Option[String] = {
+    val count = d.readUnsignedShort()
+    dbg(s"reading $count attributes")
+    if (c contains "SourceFile") {
+      val s = c("SourceFile")
+      val attributes =
+        for (_ ← 1 to count) yield {
+          val tag = d.readUnsignedShort()
+          val length = d.readInt()
+          dbg(s"  tag ${c(tag)} ($length bytes)")
+          if (tag != s) {
+            skip(d, length)
+            None
+          } else {
+            val name = d.readUnsignedShort()
+            Some(c(name))
+          }
+        }
+      dbg(s"  yielded $attributes")
+      attributes.flatten.headOption
+    } else {
+      dbg(s"  (skipped)")
+      None
+    }
+  }
+
+  private def skip(d: DataInputStream, length: Int): Unit =
+    if (d.skipBytes(length) != length) throw new IllegalArgumentException("class file ends prematurely")
+
+  private def dbg(s: String): Unit = if (debug) println("LNB: " + s)
```

Comment:
I doubt it can. It probably doesn't know if the involved `toString` methods have other side effects.